### PR TITLE
DX-276 Custom metadata changes

### PR DESCRIPTION
**Pull request by ErnestoLoma**

### CODE REVIEW

**Review by 0xquark — APPROVED** [review]

File: pyatlan/model/typedef.py [hunk 1]
```diff
@@ -233,11 +239,19 @@ def get_valid_values(self) -> Optional[list[str]]:
...
             description="Optional description of the attribute."
         )
         applicable_entity_types: Optional[str] = Field(
-            description="Set of entities on which this attribute can be applied.",
+            description="Set of entities on which this attribute can be applied. "
+            "Note: generally this should be left as-is. Any overrides should instead be applied through "
+            "one or more of applicable_asset_types}, applicable_glossary_types}, or "
+            "applicable_other_asset_types}.",
         )
         custom_applicable_entity_types: Optional[str] = Field(
```
Comment (cmgrote) [inline pyatlan/model/typedef.py:256]: Let's at least add to the description of this one that it is deprecated, to give some notice to folks that they shouldn't use this field anymore.

File: pyatlan/model/typedef.py [hunk 2]
```diff
@@ -390,6 +432,158 @@ def create(
...
+        Set of entities on which this attribute can be applied.
+        Note: generally this should be left as-is. Any overrides should instead be applied through
+        one or more of applicable_asset_types, applicable_glossary_types, or applicable_other_asset_types.
+        """
+        if self.options and self.options.applicable_entity_types:
+            return set(json.loads(self.options.applicable_entity_types))
+        return set()
+
+    @applicable_entity_types.setter
+    def applicable_entity_types(self, entity_types: set[str]):
```
Comment (cmgrote) [inline pyatlan/model/typedef.py:461]: Nice — so users can still send in a set, even though it's marshalled into JSON as a string 👍

**Review by cmgrote — CHANGES_REQUESTED** [review]
One very minor change — otherwise looks good!

File: pyatlan/model/typedef.py (hunk 2 above)
Comment (ErnestoLoma) [reply pyatlan/model/typedef.py:461]: Done

File: pyatlan/model/typedef.py (hunk 1 above)
Comment (ErnestoLoma) [reply pyatlan/model/typedef.py:256]: Done

**Review by cmgrote — APPROVED** [review]

**Pull request merged**